File: R/example_figure_formatting.R
# ------------------------------------------------------------------------------------------------
# --- Example script demonstrating figure_formatting.R functionality ---
# ------------------------------------------------------------------------------------------------

# Source the figure formatting functions
# Works when run from the R/ directory or when figure_formatting.R is in the path
if (file.exists("figure_formatting.R")) {
  source("figure_formatting.R")
} else if (file.exists("R/figure_formatting.R")) {
  source("R/figure_formatting.R")
} else {
  stop("Could not find figure_formatting.R. Please ensure it's in the current directory or R/ subdirectory.")
}

# Load required packages
library(ggplot2)
library(cowplot)

# Create output directory
output_dir <- "example_figures"
if (!dir.exists(output_dir)) {
  dir.create(output_dir)
}

# Generate sample data
x <- seq(0, 10, length.out = 100)
y <- sin(x)
data <- data.frame(x = x, y = y)

# ------------------------------------------------------------------------------------------------
# Example 1: Single plot with default font settings
# ------------------------------------------------------------------------------------------------
cat("Creating example figure with default font settings...\n")

# set figure width and height
width_mm <- 80
height_mm <- 60

# Setup figure with default settings (Helvetica equivalent, 7pt)
fig_setup <- setup_figure(width_mm = width_mm, height_mm = height_mm)

# Create the plot
p1 <- ggplot(data, aes(x = x, y = y)) +
  geom_line(linewidth = 1) +
  labs(
    x = "X axis label",
    y = "Y axis label",
    title = "Default font (sans, 7pt)"
  ) +
  fig_setup$theme_obj

# Save the figure (dimensions come from fig_setup)
save_figure(p1, file.path(output_dir, "example_default_font.pdf"),
            width_mm = width_mm, height_mm = height_mm)

# Also save as SVG
save_figure(p1, file.path(output_dir, "example_default_font.svg"),
            width_mm = width_mm, height_mm = height_mm)

# ------------------------------------------------------------------------------------------------
# Example 2: Multipanel figure (2x2 grid) with shared y axis
# ------------------------------------------------------------------------------------------------
cat("Creating multipanel figure (2x2 grid)...\n")

# Setup for multipanel figure
width_mm <- 150
height_mm <- 100
fig_setup_multi <- setup_figure(
  width_mm = width_mm,
  height_mm = height_mm,
  nrows = 2,
  ncols = 2
)

# Create individual plots with different data
plots <- list()
for (i in 1:4) {
  phase <- (i - 1) * pi / 4
  y_sub <- sin(x + phase)
  data_sub <- data.frame(x = x, y = y_sub)
  
  p <- ggplot(data_sub, aes(x = x, y = y)) +
    geom_line(linewidth = 1) +
    labs(
      x = ifelse(i >= 3, "X axis label", ""),
      y = ifelse(i %% 2 == 1, "Y axis label", ""),
      title = paste("Panel", i)
    ) +
    fig_setup_multi$theme_obj
  
  plots[[i]] <- p
}

# Combine plots using cowplot's plot_grid
combined <- plot_grid(
  plotlist = plots,
  nrow = 2,
  ncol = 2,
  align = "hv"
)

# Save the combined figure (dimensions come from fig_setup_multi)
save_figure(combined, file.path(output_dir, "example_multipanel.svg"),
            width_mm = width_mm, height_mm = height_mm)

File: Python/example_figure_formatting.py
# ------------------------------------------------------------------------------
# --- Example script demonstrating figure_formatting.py functionality ---
# ------------------------------------------------------------------------------


import sys
import os
import numpy as np
from pathlib import Path
import matplotlib.pyplot as plt
from figure_formatting_kmn import setup_figure, save_figure


# Add current directory to path to import from same directory
# Works both when run as script (__file__ exists) and interactively (uses cwd)
try:
    current_dir = Path(__file__).parent
except NameError:
    # Interactive mode: use current working directory
    current_dir = Path(os.getcwd())

# Add to path if not already there
current_dir_str = str(current_dir)
if current_dir_str not in sys.path:
    sys.path.insert(0, current_dir_str)

# Create output directory
output_dir = current_dir / 'example_figures'
output_dir.mkdir(exist_ok=True)

# Generate sample data
x = np.linspace(0, 10, 100)
y = np.sin(x)

# ------------------------------------------------------------------------------
# Example 1: Default font (Helvetica, 7pt)
# ------------------------------------------------------------------------------
print("Creating example figure with default font settings...")

fig, ax = setup_figure(width_mm=80, height_mm=60, margins_mm=(0, 0, 0, 0))
ax.plot(x, y, linewidth=1)
ax.set_xlabel('X axis label')
ax.set_ylabel('Y axis label')
ax.set_title('Default font (Helvetica, 7pt)')
save_figure(fig, output_dir / 'example_default_font.svg')
plt.close(fig)

# ------------------------------------------------------------------------------
# Example 2: Multipanel figure (2x2 grid)
# ------------------------------------------------------------------------------
print("Creating multipanel figure (2x2 grid)...")

fig, axes = setup_figure(
    width_mm=150,
    height_mm=100,
    margins_mm=(0, 0, 0, 0),
    nrows=2,
    ncols=2,
    sharex=True,  # Share x-axis across subplots
    sharey=False  # Don't share y-axis
)

# Handle axes - can be single axis, list, or numpy array
if isinstance(axes, np.ndarray):
    axes_flat = axes.flatten()
elif isinstance(axes, (list, tuple)):
    axes_flat = axes
else:
    axes_flat = [axes]

# Plot different data in each subplot
for i, ax in enumerate(axes_flat):
    x_sub = np.linspace(0, 10, 100)
    y_sub = np.sin(x_sub + i * np.pi / 4)  # Different phase for each subplot
    ax.plot(x_sub, y_sub, linewidth=1)
    ax.set_xlabel('X axis label' if i >= 2 else '')
    ax.set_ylabel('Y axis label' if i % 2 == 0 else '')
    ax.set_title(f'Panel {i+1}')

save_figure(fig, output_dir / 'example_multipanel.svg')
plt.close(fig)

print(f"\nExample figure saved to: {output_dir}")
